### PR TITLE
refactor: dialog

### PR DESCRIPTION
**Pull request by joschkabraun**
Closes https://github.com/jina-ai/now/issues/556 of https://github.com/jina-ai/now/issues/482 and additionally:

- [x] removes redundancy in `UserInput`
- [x] changes every data source (except Elasticsearch) to use `data` as the origin of the data

### CODE REVIEW

File: now/client/__init__.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def send_request(self, endpoint: str, **kwargs):
         jina_response = jina_client_post(
             app_request,
             endpoint,
-            jina_request.data,
```
Comment (azayz) [inline now/client/__init__.py:70]: why is this changing ?

File: now_common/options.py [hunk 2]
```diff
@@ -328,11 +339,6 @@ def _construct_app(app_name: str):
     )()
 
 
-@time_profiler
-def _check_requirements(user_input: UserInput, **kwargs) -> None:
-    user_input.app_instance.run_checks(user_input)
```
Comment (azayz) [inline now_common/options.py:333]: calling run_checks directly wouldn't use the decorator time profiler is this intended ?

**Review by azayz — COMMENTED** [review]
Mostly LGTM besides some few things I would like to understand more

File: now/client/__init__.py (hunk 1 above)
Comment (joschkabraun) [reply now/client/__init__.py:70]: Thanks for catching this! Got accidentally renamed.

File: now_common/options.py (hunk 2 above)
Comment (joschkabraun) [reply now_common/options.py:333]: Yes, but I don't think profiling it is actually necessary. If we want that, we can always move it to the base app. What do you think?

Comment (azayz) [reply now_common/options.py:333]: do you know the reason why we used time profiling in the first place ? If you believe that it's unnecessary then feel free to remove it as it's nit extremely important anyway

File: now_common/options.py [hunk 3]
```diff
@@ -62,23 +66,11 @@
     description='What sort of search engine would you like to build?',
 )
 
-DATA = DialogOptions(
```
Comment (azayz) [inline now_common/options.py:65]: if these changes update the cli options and dialog, can you also update the docs ?

**Review by makram93 — COMMENTED** [review]
Nice refactoring. I like it overall. My only concern is about the cli parser. @florian-hoenicke and I had a discussion long back where he suggested to have only `--data` args for cli and parse it based on the value of it. It would be nice to still have that as the only argument exposed and add a parser function that will set the `user_input` accordingly. If we wish to continue with the current changes then it will have a breaking changes in the`now_api`, docs, jina.ai website, etc
cc @florian-hoenicke

File: now_common/options.py (hunk 2 above)
Comment (joschkabraun) [reply now_common/options.py:333]: no

File: now_common/options.py (hunk 3 above)
Comment (joschkabraun) [reply now_common/options.py:65]: Good catch. Thanks!

**Comment by florian-hoenicke** [comment]
> Nice refactoring. I like it overall. My only concern is about the cli parser. @florian-hoenicke and I had a discussion long back where he suggested to have only `--data` args for cli and parse it based on the value of it. It would be nice to still have that as the only argument exposed and add a parser function that will set the `user_input` accordingly. If we wish to continue with the current changes then it will have a breaking changes in the`now_api`, docs, jina.ai website, etc cc @florian-hoenicke

@makram93 I remember our discussion. Back then I was arguing for a single --data attribute. It fits our story: just provide your data and we provide an API. 
Meanwhile, I have not a strong opinion anymore. The main usage of our product will be via setup assistance on cli and via jina cloud.
`jina now --data --...` is not something we promote at the moment. Therefore I'm happy with making our code more readable and removing redundant attributes.

**Pull request merged**